src/feature-interface.h

```diff
@@ -0,0 +1,41 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <map>
+#include <memory>
+
+
+namespace librealsense {
+
+
+// Base class for all features
+class feature_interface
+{
+public:
+    feature_interface( const std::string& id ) : _id( id )
+    {
+    }
+
+    const std::string & get_id() const { return _id; }
+
+    virtual ~feature_interface() = default;
+
+private:
+    std::string _id;
+};
+
+typedef std::map< std::string, std::shared_ptr< feature_interface > > features;
+
+// Interface for objects that has features
+class features_support
```

Comment:
Should go in its own header; definitely not here.

Reply:
Why do we need this abstract anyway? I can remove and use only `features_container`, no?

Reply:
No, the latter is an implementation of the former. See my next comment about the container.